reference/3.0/Microsoft.PowerShell.Core/About/about_Ref.md

```diff
@@ -5,72 +5,157 @@ locale:  en-us
 of a variable that is passed to it.
 
-When the [ref] type is associated with an object, it returns a reference to
-that object. If the reference is used with a method, the method can refer to
-the object that was passed to it. If the object is changed within the method,
-the change appears as a change in the value of the variable when control
-returns to the calling method.
+## Long description
 
-To use referencing, the parameter must be a reference variable. If it is not,
-an InvalidArgument exception is thrown.
+You can pass variables to methods *by reference* or *by value*.
 
-The parameters used in method invocations must match the type required by the
-methods.
+When you pass a variable *by value*, you are passing a copy of the data.
 
-Examples:
+In the following example, the function changes the value of the variable passed
+to it. In PowerShell, integers are value types so they are passed by value.
+Therefore, the value of `$var` is unchanged outside the scope of the function.
 
 ```powershell
-PS> function swap([ref]$a,[ref]$b)
->> {
->>     $a.value,$b.value = $b.value,$a.value
->> }
-
-PS> $a = 1
-PS> $b = 10
-PS> $a,$b
-1
-10
-PS> swap ([ref]$a) ([ref]$b)
-PS> $a,$b
+Function Test($data)
+{
+    $data = 3
+}
+
+$var = 10
+Test -data $var
+$var
+```
+
+```output
 10
-1
+```
+
+In the following example, a variable containing a `Hashtable` is passed to a
+function. `Hashtable` is an object type so by default it is passed to the
+function *by reference*.
+
+When passing a variable *by reference*, the function can change the data and
+that change persists after the function executes.
+
+```powershell
+Function Test($data)
+{
+    $data.Test = "New Text"
+}
+
+$var = @{}
+Test -data $var
+$var
+```
+
+```output
+Name                           Value
+----                           -----
+Test                           New Text
+```
+
+The function adds a new key-value pair that persists outside of the function's
+scope.
+
+### Writing functions to accept reference parameters
+
+You can code your functions to take a parameter as a reference, regardless of
+the type of data passed. This requires that you specify the parameters type
+as `System.Management.Automation.PSReference`, or `[ref]`.
+
+When using references, you must use the `Value` property of the
+`System.Management.Automation.PSReference` type to access your data.
+
+```powershell
+Function Test([ref]$data)
+{
+    $data.Value = 3
+}
+```
+
+To pass a variable to a parameter that expects a reference, you must type
+cast your variable as a reference.
 
-PS C:\ps-test> function double
->> {
->>     param ([ref]$x) $x.value = $x.value * 2
->> }
+> [!NOTE]
+> The brackets and parenthesis are BOTH required.
 
-PS> $number = 8
+```powershell
+$var = 10
+Test -data ([ref]$var)
+$var
+```
+
+```output
+3
+```
+
+### Passing references to .NET methods
+
+Certain .NET methods may require you to pass a variable as a reference. When
```

Comment:
I meant to comment this one last time.

Some .NET methods require you to pass a variable by reference.

Reply:
fixed